plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx

```diff
@@ -12,6 +18,44 @@ import { ProductSectionLayout } from '../layout/product-section-layout';
 import { Variations } from '../fields/variations';
 
 export const ProductVariationsSection: React.FC = () => {
+	const {
+		getInputProps,
+		values: { id: productId },
+	} = useFormContext< Product >();
+
+	const { value: attributes }: { value: ProductAttribute[] } = getInputProps(
+		'attributes',
+		{
+			productId,
+		}
+	);
+
+	const options = attributes
+		? attributes.filter(
+				( attribute: ProductAttribute ) => attribute.variation
+		  )
+		: [];
+
+	const { totalCount } = useSelect(
+		( select ) => {
+			const { getProductVariationsTotalCount } = select(
+				EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+			);
+			const requestParams = {
+				product_id: productId,
+			};
+			return {
+				totalCount:
+					getProductVariationsTotalCount< number >( requestParams ),
+			};
+		},
+		[ productId, options ]
+	);
+
+	if ( options.length === 0 || totalCount === 0 || isNaN( totalCount ) ) {
```

Comment:
I think we can remove all of this `totalCount` logic now and just rely on the `options.length`.

On a new product page when adding options, this `totalCount` will not get updated and the variations will remain hidden.

Reply:
Thank you @joshuatf for your review.

This is interesting; when I place a breakpoint [here](https://github.com/woocommerce/woocommerce/pull/36202/files/d88d91cfda8661b7fe5e1b0ac64c34b72f56dc86#diff-f0bcdfc13aba51d445b30531587b77a929f1e47ca619b9cd4d53e87df0595735R41) the `totalCount`const is loaded correctly after adding a new option. I guess that there is something related to the variations creation time.

As you mentioned, it seems like we can remove the `totalCount`. I removed that logic in the commit [e9d7f4d](https://github.com/woocommerce/woocommerce/pull/36202/commits/e9d7f4d232b6f1c1fb6f8a051002c14d53c39c42)